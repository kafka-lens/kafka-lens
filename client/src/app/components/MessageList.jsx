import React from 'react';
import Message from './Message.jsx';
import '../css/MessageList.scss';

const MessageList = ({ messageArray }) => {
  const renderMessages = [];

  messageArray.forEach((msg, i) => {
    renderMessages.push(
      <Message key={i} id={i} message={msg.value} offset={msg.offset} />,
    );
  });

  return <div className="message-list">{renderMessages}</div>;
};

export default MessageList;
