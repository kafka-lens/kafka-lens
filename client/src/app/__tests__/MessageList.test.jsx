import React from 'react';
import { shallow } from 'enzyme';

import MessageList from '../components/MessageList';

describe('MessageList.js unit tests', () => {
  let wrapper;
  const props = {
    messageArray: [],
    topicName: 'first',
  };

  beforeAll(() => {
    for (let i = 0; i < 5; i++) {
      props.messageArray.push({ offset: i, message: i });
    }
    wrapper = shallow(<MessageList {...props} />);
  });

  it('Should render without errors', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render message components', () => {
    expect(wrapper.exists('Message')).toBe(true);
  });
});
