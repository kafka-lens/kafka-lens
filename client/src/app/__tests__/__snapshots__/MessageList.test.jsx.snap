// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageList.js unit tests Should render without errors 1`] = `
<div
  className="message-list"
>
  <Message
    id={0}
    key="first-0"
    offset={0}
  />
  <Message
    id={1}
    key="first-1"
    offset={1}
  />
  <Message
    id={2}
    key="first-2"
    offset={2}
  />
  <Message
    id={3}
    key="first-3"
    offset={3}
  />
  <Message
    id={4}
    key="first-4"
    offset={4}
  />
</div>
`;
